### DISCUSSION TITLE
Upload image inside a form without sending whole form

### DISCUSSION THREAD

**Discussion by Eiley2** [remix-run/remix#6527] [Q&A] [answered]
Hi

I'm kinda new to remix and web dev, so excuse me if this is a noob question. I having troubles with this situation:

I have a form to create a new ClientProfile inside my app. Inside the form I need to allow the user to upload an image to S3 or wherere I can show it in the page, and visualize it inside an img tag before sending the whole form. I'm having troubles deciding how to do that. Should I use a fetcher inside the Form? 

Also, if anyone has a better idea for managing the form, it's welcomed.

**Comment by sadin999** [comment]
I think this example answers what you are after:

https://github.com/remix-run/examples/tree/main/file-and-s3-upload

**Reply by Eiley2** [reply]
Well, not really.

I mean I have this:

`
<Form method="post" className="mt-4 flex flex-col items-center gap-y-2 self-center rounded-xl bg-white px-4 py-8 sm:w-[500px] lg:w-[750px]">

        <input
          type="text"
          ref={titleRef}
          name="name"
          id="name"
          className="mb-5 w-[100%] max-w-[450px] border-b-2 bg-transparent text-center text-base font-bold text-zinc-700 outline-none sm:text-xl"
          placeholder="Nombre..."
          onClick={() => titleRef.current?.select()}
        />
        <img
          ref={profilePicRef}
          src="/img/PlayerPlaceholder.jpg"
          className="w-60 rounded"
          alt="imagen"
        />
        <input
          id="img-field"
          type="file"
          name="profilePic"
          accept="image/*"
          className="text-xs"
        />
      </Form>`

I need to automatically upload the img-field without submitting the whole form.

**Reply by kiliman** [reply, marked as answer]
Take a look at this article. If you sign up for his newsletter, you'll have access to the code.

https://www.jacobparis.com/content/remix-image-uploads

**Reply by jacobparis** [reply]
I've open sourced this to everyone now so you can check it out here

https://github.com/jacobparis-insiders/jacobparis.com/tree/main/app/examples/remix-image-uploads
